backend/inspirehep/records/api/literature.py

```diff
@@ -317,6 +318,27 @@ def add_figures(self, figures):
     def add_file(
         self, url, original_url=None, key=None, filename=None, *args, **kwargs
     ):
+        if current_s3_instance.is_s3_url(url) and not current_app.config.get(
+            "UPDATE_S3_FILES_METADATA", False
+        ):
+            if key not in url:
+                filename = filename or key
+                key = url.split("/")[-1]
+            result = {
+                "key": key,
+                "filename": filename,
+                "url": url,
+                "original_url": original_url,
+                **kwargs,
```

Comment:
I know this is not related to your PR but I just realized we are also returning mimetype and size and these aren't on the schema nor added by the builder so they are returned for nothing, could you remove those too please?

Reply:
:+1: